### PR TITLE
Add signup

### PR DESCRIPTION
**Pull request by zhangium**
Branch should really be called add signup. Generated the users controller and made a view for it. Ugly for now but it works. 🔥

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,11 +1,17 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>Sydedoor</title>
+    <title>SydeDoor</title>
     <%= csrf_meta_tags %>
 
+    <%# Commented the block below because tests were failing%>
```
Comment (ryan-mok) [inline app/views/layouts/application.html.erb]: Why were the tests failing? What was the error?

Comment (zhangium) [reply app/views/layouts/application.html.erb]: `UsersControllerTest#test_should_get_new:
ActionView::Template::Error: TypeError: Object doesn't support this property or method
    app/views/layouts/application.html.erb:7:in `_app_views_layouts_application_html_erb__575790912_48294960'
    test/controllers/users_controller_test.rb:5:in `block in <class:UsersControllerTest>'

C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/test_unit/reporter.rb:70:in `method': undefined method `test_should_get_new' for class `Minitest::Result' (NameError)
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/test_unit/reporter.rb:70:in `format_rerun_snippet'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/test_unit/reporter.rb:23:in `record'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:803:in `block in record'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:802:in `each'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:802:in `record'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:334:in `run_one_method'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:321:in `block (2 levels) in run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:320:in `each'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:320:in `block in run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:360:in `on_signal'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:347:in `with_info_handler'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:319:in `run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/test_unit/line_filtering.rb:9:in `run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:159:in `block in __run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:159:in `map'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:159:in `__run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:136:in `run'
        from C:/RailsInstaller/Ruby2.3.3/lib/ruby/gems/2.3.0/gems/minitest-5.11.3/lib/minitest.rb:63:in `block in autorun'
`

File: Gemfile [hunk 2]
```diff
@@ -9,6 +9,7 @@ gem 'coffee-rails', '4.2.2'
 gem 'jquery-rails', '4.3.1'
 gem 'turbolinks',   '5.0.1'
 gem 'jbuilder',     '2.7.0'
+gem 'rails-controller-testing'
```
Comment (ryan-mok) [inline Gemfile]: What's this gem for?

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -12,4 +12,4 @@
 //
 //= require rails-ujs
 //= require turbolinks
-//= require_tree .
+//= require_tree . unless
```
Comment (ryan-mok) [inline app/assets/javascripts/application.js]: What does `unless` do in this context

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+
```
Comment (ryan-mok) [inline app/controllers/users_controller.rb]: Tab spacing

File: Gemfile (hunk 2 above)
Comment (zhangium) [reply Gemfile]: for controller testing!

File: app/assets/javascripts/application.js (hunk 3 above)
Comment (zhangium) [reply app/assets/javascripts/application.js]: my mistake, will resolve

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,11 +1,14 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>Sydedoor</title>
+    <title>SydeDoor</title>
     <%= csrf_meta_tags %>
 
-    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
```
Comment (ryan-mok) [inline app/views/layouts/application.html.erb]: Why was `'data-turbolinks-track': 'reload'` for `stylesheet_link_tag` removed?

**Pull request merged**